src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -1009,6 +1009,15 @@ object Parsers {
               in.nextToken()
               expr()
             } else EmptyTree
+
+          handler match {
+            case Block(Nil, EmptyTree) => syntaxError(
+              "`catch` block does not contain a valid expression, try adding a case like - `case t: Throwable =>` to the block",
```

Comment:
Ha you're right, I googled to find the doc and didn't realize I was looking at the 2.9 version :) http://www.scala-lang.org/api/2.9.3/scala/util/control/NonFatal$.html